### PR TITLE
docs(bundling): add rspack to docs

### PR DESCRIPTION
**Pull request by jaysoo**
**⚠️ Don't merge this PR yet, we're waiting until March 9th 6:30 PM EST to merge, but we need approvals prior to that time.**

This PR adds the rspack docs to nx.dev. It is currently in the labs repo so it cannot be handled the same as the other packages.

We're keeping it self-contained in `nx-dev/nx-dev/pages/packages/rspack` so it can be replaced/deleted later with much work. There are three main changes:

1. Add `rspack` pages (can be easily deleted later).
2. Update sidebar component to splice in `rspack` into menu.
3. Update doc viewer component to hide "Edit" button when the packege is not in repo. This could be done better by supporting external repos, but it's a workaround that will be removed later, so not worth enhancing.

Preview: https://nx-dev-git-fork-jaysoo-docs-add-rspack-nrwl.vercel.app/packages/rspack
Screenshot 2023-03-08 at 3 47 18 PM
Screenshot 2023-03-08 at 3 47 42 PM

### CODE REVIEW

File: nx-dev/nx-dev/lib/rspack/content/overview.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+\`\`\`shell
+npx create-nx-workspace@latest --preset=@nrwl/rspack
+\`\`\`
+
+You will be prompted for a repository name, which will be used for the folder and application name.
+
+## Add Rspack to an existing workspace
+
+There is a number of ways to use Rspack in your existing workspace.
```
Comment (isaacplmann) [inline nx-dev/nx-dev/lib/rspack/content/overview.ts:28]:
```suggestion
There are a number of ways to use Rspack in your existing workspace.
```

File: nx-dev/nx-dev/lib/rspack/content/overview.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+Finally, configure Rspack for the new project.
+
+\`\`\`bash
+nx g @nrwl/rspack:configuration --project=my-app
+\`\`\`
+
+### Modify an existing React or Web project to use Rspack
+
+You can use the \`@nrwl/vite:configuration\` generator to change your React or Web project to use Rspack.
```
Comment (isaacplmann) [inline nx-dev/nx-dev/lib/rspack/content/overview.ts:89]:
```suggestion
You can use the \`@nrwl/rspack:configuration\` generator to change your React or Web project to use Rspack.
```

File: nx-dev/nx-dev/lib/rspack/content/overview.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+This generator will modify your project's configuration to use Rspack, and it will also install all the necessary dependencies.
+
+You can read more about this generator on the [\`@nrwl/vite:configuration\`](/packages/vite/generators/configuration) generator page.
```
Comment (isaacplmann) [inline nx-dev/nx-dev/lib/rspack/content/overview.ts:92]:
```suggestion
You can read more about this generator on the [\`@nrwl/rspack:configuration\`](/packages/rspack/generators/configuration) generator page.
```

**Review by isaacplmann — CHANGES_REQUESTED** [review]
Some typos

**Review by isaacplmann — APPROVED** [review]

**Review by bcabanes — APPROVED** [review]

**Pull request merged**